.github/workflows/tests.yml

```diff
@@ -37,7 +40,43 @@ jobs:
             const { issue: { number: issue_number }, repo: { owner, repo } } = context;
             const label = 'run-tests';
             github.issues.removeLabel({ owner, repo, issue_number, name: label });
-
+      - name: Checkout Amplify UI
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ steps.get-ref.outputs.ref }}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          persist-credentials: false
+      - name: Setup Node.js LTS
+        uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+          cache: 'yarn'
+      - name: Get ref and commit
+        id: get-ref
+        run: |
+          echo "::set-output name=key::${{ github.event_name == 'push' && github.ref || github.event.pull_request.head.ref }}"
+          echo "::set-output name=commit::$(git rev-parse --short "$GITHUB_SHA")"
+      - name: Install packages
+        run: yarn --frozen-lockfile --prefer-offline
+      - name: Build ui package
+        run: yarn ui build
+      - name: Cache cypress runner
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/Cypress
+          key: ${{ runner.os }}-cypress-${{ hashFiles('yarn.lock') }}
+      - name: Cache packages/ui/dist
+        uses: actions/cache@v2
+        with:
+          path: ./packages/ui/dist
+          key: ${{ runner.os }}-ui-${{ steps.get-ref.outputs.commit }}
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: |
+            ./node_modules
+            **/node_modules
```

Comment:
This would be the first cache to remove if we're facing any cache misses. `node_modules` is large at ~600mb, and `yarn` is already being cached, so this doesn't save that much time for its storage cost.

For reference, based on https://github.com/aws-amplify/amplify-ui/actions/runs/1560169048, caching `node_modules` saved 47 seconds. If we assume we implement conditional testing and only run unit testing, that goes down to only 7 seconds. I'll revisit this when we do conditional testing.

Reply:
`actions/cache` is very generous though and provides 10GB of cache storage. https://github.com/actions/cache

Reply:
Actually, screw it, I'll get rid of it. This is really a duplciate of `yarn cache`, I'm down to have conditional testing do more work than this tiny optimization.

Reply:
(sry for this long thread, I was going back and forth)